.github/workflows/checkcode.yml

```diff
@@ -20,4 +20,41 @@ jobs:
           source ./utilities/python-venv/bin/activate
           pre-commit clean
           pre-commit run --all-files --show-diff-on-failure
+  codeclimate:
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/ppraegla/4c-dependencies:latest
+      options: --user root --env OMPI_ALLOW_RUN_AS_ROOT=1 --env OMPI_ALLOW_RUN_AS_ROOT_CONFIRM=1
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ secrets.github_token }}
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/configure_4C
+        with:
+          cmake-preset: docker_codeclimate
+          build-directory: ${{ github.workspace }}/../clang-tidy-build
+      - name: Run Clang-tidy on codebase
+        run: |
+          cd $GITHUB_WORKSPACE/../clang-tidy-build
+          mkdir -p clang-tidy-issues
+          ESCAPED_PROJECT_DIR=$(echo "${GITHUB_WORKSPACE}" | sed 's/\//\\\//g')
+          run-clang-tidy -j `nproc` -config-file ${GITHUB_WORKSPACE}/.clang-tidy -use-color -p . -export-fixes ./clang-tidy-issues/ -header-filter "${ESCAPED_PROJECT_DIR}\/.*" 2>&1 | python ${GITHUB_WORKSPACE}/utilities/code_climate/filter_clang_tidy_output.py
+      - name: Export Clang tidy issues
+        run: |
+          python3 -m venv venv
+          . venv/bin/activate
+          pip install wheel
+          pip install -r ${GITHUB_WORKSPACE}/utilities/code_climate/requirements.txt
+          python ${GITHUB_WORKSPACE}/utilities/code_climate/export_clang_tidy_issues.py -j `nproc` --code-climate-file ${GITHUB_WORKSPACE}/codeclimate.json --source-folder ${GITHUB_WORKSPACE} ${GITHUB_WORKSPACE}/../clang-tidy-build/clang-tidy-issues/*.yaml
+      - name: Upload clang-tidy issues
+        uses: actions/upload-artifact@v4
+        with:
+          name: codeclimate-report
+          path: |
+            codeclimate.json
```

Comment:
Can Github use this code climate file the same way as Gitlab does? I couldn't find anything from a quick search.

Reply:
No. There are automatic actions that add comments to the PR, but I don't think that is feasible here for us as we still have too many warnings. We could also remove the artifact.

Reply:
Just a curiosity. We should probably convert a lot more clang-tidy warnings into errors in the future. The report itself was never that useful to me on Gitlab anyway.